test/uri/uri_recompose_test.cc

```diff
@@ -33,5 +33,5 @@ TEST(URI_recompose, no_scheme) {
 
 TEST(URI_recompose, empty_fragment) {
   const sourcemeta::jsontoolkit::URI uri{"#"};
-  EXPECT_EQ(uri.recompose(), "#");
+  EXPECT_EQ(uri.recompose(), "");
```

Comment:
⚠️ @jviotti

Reply:
I think we should fix this one too. `"#"` is URI with an empty fragment, while `""` is an empty URI, which is not the same

Reply:
Done